### PR TITLE
Change the custom config parser with the oslo.config

### PR DESCRIPTION
**Pull request by alexcoman**
For the moment argus framework is using a custom config
parser which is hard to extend and maintain. In order to
avoid redundant work the old config parser it will be
changed with oslo.config.

### CODE REVIEW

**Review by ader1990 — APPROVED** [review]
lgtm

File: argus/config/opts.py [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+"""
+This is the single point of entry to generate the sample configuration
+file for Cloudbase-Init.
```
Comment (stefan-caraiman) [inline argus/config/opts.py]: Cloudbase-Init-CI

File: argus/config/cloudbaseinit.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+from argus.config import base as conf_base
+
+
+class CloudbaseInitOptions(conf_base.Options):
+
+    """Config options available for the cloudbase-init setup."""
+
+    def __init__(self, config):
+        super(CloudbaseInitOptions, self).__init__(config, group="argus")
```
Comment (stefan-caraiman) [inline argus/config/cloudbaseinit.py]: `group="cloudbaseinit"`

File: argus/config/openstack.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+from argus.config import base as conf_base
+
+
+class OpenStackOptions(conf_base.Options):
+
+    """Config options available for the openstack setup."""
+
+    def __init__(self, config):
+        super(OpenStackOptions, self).__init__(config, group="argus")
```
Comment (stefan-caraiman) [inline argus/config/openstack.py]: `group="openstack"`

File: argus/config/openstack.py [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+                help="The id of the image that is to be used for tests."),
+            cfg.StrOpt(
+                "flavor_ref", default="3",
+                help="The id of the flavor that is to be used."),
+            cfg.StrOpt(
+                "image_username", default="CiAdmin",
+                help="The default username which can connect to the instance. "
+                     " It should be created when the image is created."),
+            cfg.StrOpt(
+                "image_password", default="P@ssw0rd",
```
Comment (stefan-caraiman) [inline argus/config/openstack.py]: `default="Passw0rd"`

File: argus/config/cloudbaseinit.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        self._options = [
+            cfg.StrOpt("created_user", default=None),
```
Comment (stefan-caraiman) [inline argus/config/cloudbaseinit.py]: The defaults were `Admin` and group `Administrators`, but it's not a deal breaker.

File: argus/config/ci.py [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+class ArgusOptions(conf_base.Options):
+
+    """Config options available for the argus framework setup."""
+
+    def __init__(self, config):
+        super(ArgusOptions, self).__init__(config, group="argus")
+        self._options = [
+            cfg.StrOpt(
+                "resources", default=_RESOURCES_LINK,
+                help="A url that holds the resources usualy from "
```
Comment (stefan-caraiman) [inline argus/config/ci.py]: An url_, usually_

File: argus/config/ci.py [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+                     "/argus/resources avalible on the web"),
```
Comment (stefan-caraiman) [inline argus/config/ci.py]: Available

File: argus/config/openstack.py [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+            cfg.StrOpt("image_os_type", default=None),
+            cfg.BoolOpt(
+                "require_sysprep", default=True,
+                help="If the provided image require the sysprep before start "
+                     "running tests"),
+        ]
+
+    def register(self):
+        """Register the current options to the global ConfigOpts object."""
+        group = cfg.OptGroup(self.group_name, title='Cloudbase-Init Options')
```
Comment (stefan-caraiman) [inline argus/config/openstack.py]: OpenStack*

**Review by stefan-caraiman — COMMENTED** [review]
Conflicts

**Pull request merged**